app/src/main/java/org/listenbrainz/android/repository/listenservicemanager/ListenServiceManagerImpl.kt

```diff
@@ -40,7 +41,7 @@ class ListenServiceManagerImpl @Inject constructor(
     /** Used to avoid repetitive submissions.*/
     private var lastNotificationPostTs = System.currentTimeMillis()
     private lateinit var whitelist: List<String>
-    private var isScrobblingAllowed: Boolean = true
+    private var isListeningAllowed: AtomicBoolean = AtomicBoolean(true)
```

Comment:
I see you are trying to make this thread safe, but why was this necessary? Can you please link the issue this solves?
Also, why is it being solved in this PR. Totally irrelevant

Reply:
Replied on IRC.